src/Nri/Ui/Highlighter/V1.elm

```diff
@@ -302,74 +302,94 @@ update msg model =
                 ( model, Cmd.none )
 
 
-nextInteractive : Int -> List (Highlightable marker) -> Int
-nextInteractive index highlightables =
-    if Maybe.map .type_ (List.Extra.getAt index highlightables) == Just Highlightable.Interactive then
-        index
+nextInteractiveIndex : Int -> List (Highlightable marker) -> Int
+nextInteractiveIndex index highlightables =
+    let
+        isInteractive highlightable =
+            .type_ highlightable == Highlightable.Interactive
 
-    else if index < List.length highlightables - 1 then
-        nextInteractive (index + 1) highlightables
+        interactiveHighlightables =
+            List.filter isInteractive highlightables
+    in
+    List.foldl
+        (\x ( nextIndex, hasIndexMatched ) ->
+            if hasIndexMatched then
+                ( x.groupIndex, False )
 
-    else
-        index
+            else
+                ( nextIndex, x.groupIndex == index )
+        )
+        ( index + 1, False )
+        interactiveHighlightables
+        |> Tuple.first
 
 
-previousInteractive : Int -> List (Highlightable marker) -> Int
-previousInteractive index highlightables =
-    if Maybe.map .type_ (List.Extra.getAt index highlightables) == Just Highlightable.Interactive then
-        index
+previousInteractiveIndex : Int -> List (Highlightable marker) -> Int
+previousInteractiveIndex index highlightables =
+    let
+        isInteractive highlightable =
+            .type_ highlightable == Highlightable.Interactive
 
-    else if index > 0 then
-        previousInteractive (index - 1) highlightables
+        interactiveHighlightables =
+            List.filter isInteractive highlightables
+    in
+    List.foldr
+        (\x ( nextIndex, hasIndexMatched ) ->
+            if hasIndexMatched then
+                ( x.groupIndex, False )
 
-    else
-        index
+            else
+                ( nextIndex, x.groupIndex == index )
+        )
+        ( index - 1, False )
```

Comment:
I know we discussed not having a Maybe here, but I think I thought of an edge case, unfortunately!

Suppose our highlightables are something like:

```
Static 0 | Static 1 | Static 2 | Interactive 3
```

If we're focused on Interactive 3, and we hit the left arrow, I think we'll end up on Static 2. And actually we should stay on Interactive 3.

I'm thinking that the out-of-bounds checks will need to move into `nextInteractive` and `previousInteractive`, and we'll need to use a Maybe. 🤔